website_sale_menu_partner_top_selling/controllers/main.py

```diff
@@ -0,0 +1,110 @@
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.website_sale.controllers.main import TableCompute, WebsiteSale
+
+
+class WebsiteSale(WebsiteSale):
+    @http.route(
+        ["/shop/my_regular_products", "/shop/my_regular_products/page/<int:page>"],
+        type="http",
+        auth="public",
+        website=True,
+    )
+    def user_regular_products(self, page=0, ppg=False, **kwargs):
+        if request.env.user.has_group("base.group_public"):
+            return request.redirect("/web/login")
+        param_limit = int(
+            request.env["ir.config_parameter"]
+            .sudo()
+            .get_param("website_sale_menu_partner_top_selling.limit", 10)
+        )
+        # Get best-selling products from the user
+        partner = request.env.user.partner_id
+        company_partner = partner.commercial_partner_id or partner
+        product_data = (
+            request.env["sale.order.line"]
+            .sudo()
+            .read_group(
+                [
+                    (
+                        "order_id.partner_id.commercial_partner_id",
+                        "=",
+                        company_partner.id,
+                    ),
+                    ("order_id.state", "in", ["sale", "done"]),
+                    ("product_id.product_tmpl_id.website_published", "=", True),
+                    "|",
+                    ("product_id.product_tmpl_id.website_id", "=", request.website.id),
+                    ("product_id.product_tmpl_id.website_id", "=", False),
+                    "|",
+                    (
+                        "product_id.product_tmpl_id.company_id",
+                        "=",
+                        request.website.company_id.id,
+                    ),
+                    ("product_id.product_tmpl_id.company_id", "=", False),
```

Comment:
I think this can be simplified to:
```suggestion
                    ("product_id.product_tmpl_id.website_id", "in", [request.website.id, False]),
                    (
                        "product_id.product_tmpl_id.company_id",
                        "in",
                        [request.website.company_id.id, False],
                    ),
```